src/sync/impl/sync_client.hpp

```diff
@@ -42,6 +46,24 @@ struct SyncClient {
         client.run();
     }) // Throws
     {
+#if REALM_PLATFORM_APPLE
```

Comment:
What's stopping us from having an implementation of `NetworkReachabilityObserver` that calls into the binding to provide the required functionality? The Java binding could provide an implementation, and .NET would provide an implementation using whatever API they have available to them.